apis/datasciencecluster/v1/datasciencecluster_types.go

```diff
@@ -80,10 +81,18 @@ type Components struct {
 
 // ComponentsStatus defines the custom status of DataScienceCluster components.
 type ComponentsStatus struct {
-	// ModelRegistry component status
-	ModelRegistry *componentsv1.DSCModelRegistryStatus `json:"modelregistry,omitempty"`
+	ModelRegistry        *componentsv1.ModelRegistryStatus `json:"modelregistry,omitempty"` // with namespace
+	Dashboard            *componentsv1.DashboardStatus     `json:"dashboard,omitempty"`     // with URL
+	CodeFlare            *c.Status                         `json:"codeflare,omitempty"`
+	Ray                  *c.Status                         `json:"ray,omitempty"`
+	TrustyAI             *c.Status                         `json:"trustyai,omitempty"`
+	TrainingOperator     *c.Status                          `json:"trainingoperator,omitempty"`
+	Kueue                *c.Status                         `json:"kueue,omitempty"`
+	DataSciencePipelines *c.Status                         `json:"datasciencepipelines,omitempty"`
+	Kserve               *c.Status                         `json:"kserve,omitempty"`
+	ModelMeshServing     *c.Status                         `json:"modelmeshserving,omitempty"`
+	Workbenches          *c.Status                         `json:"workbenches,omitempty"`
```

Comment:
I don't think we should copy the entire status, only a shared portion like what we do for the spec. Probably it should be something to add to the component interface

Reply:
I would expect to have something like 

```go
type DashboardCommonStatus struct {
	URL string `json:"url,omitempty"`
}


type DashboardStatus struct {
	components.Status `json:",inline"`
        DashboardCommonStatus `json:",inline"`
}
```



```go
func (s *componentHandler) UpdateStatus(
    dsc *dscv1.DataScienceCluster, 
    c client.Object,
) {
	dsc.Status.Component.Dashbaord = c.(*compoenntsv1.Dashbaord).Status.DashboardCommonStatus
}
```